### ISSUE TITLE
Metin 2 - Terrain textures "drifting" around camera view

### ISSUE THREAD

**Issue by xKrasheR** [NVIDIAGameWorks/rtx-remix#283]
**Describe the bug**
- Terrain textures "drifting" based on camera view when enabling Runtime Terrain Baking and categorizing all terrain textures beforehand
- dxvk-remix-458b010-347-release
- bridge-remix-1126a3c-40-release
- RTX 3090
- 546.31
- Metin 2 private server build "InsomniaMT2" - Download/Registration available here https://insomniamt2.pl/ (using a translator should help)

**Attach files!**
- [NvRemixBridge.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/13629290/NvRemixBridge.log)
- [d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/13629291/d3d9.log)
- [InsomniaMT2_d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/13629301/InsomniaMT2_d3d9.log)
- [rtx.conf.txt](https://github.com/NVIDIAGameWorks/rtx-remix/files/13629303/rtx.conf.txt)

**To Reproduce**
1. Download the game client from a hosting of choice https://insomniamt2.pl/download
2. Register a game account on https://insomniamt2.pl/account/register
3. Launch _Autopatcher_InsomniaMT2.exe to patch the game first
4. Install RTX Remix
5. Open InsomniaMT2.exe to launch the game
6. Upon launching the game, you will have to mark some textures as UI in order to see anything
7. Enter your account credentials to login, create a character (Stwórz Postać) and enter the game (Wybierz Postać button or Enter key)
8. Straight after loading into the game, press TAB to bring up a teleportation menu and choose "Poczekalnia Dungeonów" level (smallest map in the game, so it is easy to categorize all the needed textures)
9. Make necesarry Game Setup steps (Categorize sky, UI and most importantly, terrain textures)
10. In Parameter Tuning, make sure to have Scene Z-Up enabled
11. Enable Runtime Terrain Baking, move the camera (R key zooms in, F key zooms out, Right Mouse Button to rotate the view)

**Expected behavior**
- Textures staying in place despite various camera angles (rasterized video for reference below)

**Screenshots**
[Video of the mentioned issue on RTX Remix](https://youtu.be/-tKVus55Ghc)
[Rasterized/Vanilla game](https://youtu.be/5doQe7KpQUo)

**Comment by MarkEHenderson** [comment]
REMIX-1778 for tracking

**Comment by NV-DM** [comment]
Updating tracking: REMIX-2655

**Comment by caiocinel** [comment]
This problem also happens in Postal 2 (which runs on Unreal Engine 2)

If "Scene Z-Up" is checked, when increasing the camera height things remain in the same place (which is correct), if it is deactivated the textures will also follow the camera height.

**Comment by Kim2091** [comment]
Also happens in Flatout 1 and 2, and occasionally in Skyrim

**Comment by Arrokoth7323** [comment]
Happens in Star Wars Empire at War (2006) 

[![**Streamable Video Link**]](https://streamable.com/qymarg?src=player-page-share)

**Comment by automata-rtx** [comment]
Confirmed getting the same issue in Star Wars Republic Commando. Unreal Engine 2 game

**Comment by UncleRito** [comment]
Happens in delta force black hawk down as well

**Comment by RJP1992** [comment]
Also happens for Battlefront II on almost every map that uses animated water terrain/ground terrain (except Tatooine for some reason).

**Comment by sirzento** [comment]
Still happens in 1.2.4. It does also occure on any metin 2 build, not just this private server.

**Comment by 318ti** [comment]
> Still happens in 1.2.4. It does also occure on any metin 2 build, not just this private server.

Thats correct.

**Comment by MarkEHenderson** [comment]
Hi,

We've added a `terrain as decals` mode (`rtx.terrainBaker.enableBaking = false`), which blends terrain layers by treating the upper layers as decals.  Please retest these games using that mode, and report back if there are still problems.  You may need to remove the 'terrain' category flag from whichever layer of terrain is drawn first for it to work properly.

As an alternative workaround, adding `rtx.terrainBaker.clearTerrainBeforeBaking = True` to your rtx.conf may help the terrain baker function correctly, so please test that too.

(And of course, please do these on a recent version of the runtime).

**Comment by NV-LL** [comment]
Hi @318ti - checking in to see if you've been able to troubleshoot this issue per the previous comment. We'd also like to inform you of our stale ticket policy, specifically that will close this ticket as 'went stale' if it does not receive an update in the next 2 weeks. Thank you!

**Comment by 318ti** [comment]
> Hi [@318ti](https://github.com/318ti) - checking in to see if you've been able to troubleshoot this issue per the previous comment.

Yes, nothing has changed since this issue has been created. Setting clearTerrainBeforeBaking to true doesn't improve anything, neither does "terrain as decals" mode as shown in the videos below, checked on the latest runtime version.

Also please take a look at this RTX Remix discord server thread where i described necessary repro details with one of the Remix developers over 2 years ago - https://discord.com/channels/1028444667789967381/1199049373502410904

https://github.com/user-attachments/assets/c6ff299c-e676-496d-b5c1-6cd7549e7651

https://github.com/user-attachments/assets/7a75e1f1-ea30-4197-8fc3-2f08ed46dc7e

https://github.com/user-attachments/assets/a6aa688b-c5bb-4062-b9e2-a18cf541b331

**Comment by NV-LL** [comment]
Hey @318ti - thanks for the update. We have a ticket open (REMIX-2655) to address terrain baking issues; in the meantime our best recommendation is to play around with different combinations of tagging/untagging terrain layers, especially the base layer. Otherwise we will keep you posted as we have updates on the internal ticket.
